kyaml/runfn/runfn.go

```diff
@@ -313,6 +313,14 @@ func (r RunFns) getFunctionFilters(global bool, fns ...*yaml.RNode) (
 			// resource doesn't have function spec
 			continue
 		}
+		// convert declarative relative paths to absolute (otherwise docker will throw an error)
```

Comment:
Some justification for doing this:

- it makes sense to allow our declarative mount paths in kustomize to be relative paths because we treat the kustomization as a single portable configuration that in theory should be able to be moved to different directories
- the reasons that Docker doesn't allow bind mount paths to be relative are [described here](https://github.com/docker/cli/issues/1203#issuecomment-981757086), and it seems mostly like this is because bind mounts are performed on the daemon side. I don't think that means that we should disallow relative paths on the kustomize side if conversion is easy for us to do.

Reply:
I agree with allowing paths to be relative (in fact I'm thinking we should require them to be), and I understand the need to convert them to absolute for Docker. But don't we need to do that at some point where we have the Kustomization root? Or else we're expanding them relative to the current working directory, I assume. I think we should treat relative paths here consistently with what we decided for exec paths.

Reply:
> But don't we need to do that at some point where we have the Kustomization root?

Good catch! I've updated the PR so that the working directory gets propagated correctly down to the containerized function filter. I've done manual testing to make sure that the path is resolved against the kustomization root.